### DISCUSSION TITLE
How does ssh keep alive?

### DISCUSSION THREAD

**Discussion by isPoto** [kingToolbox/WindTerm#568] [Q&A] [answered]
ssh will automatically disconnect if it is idle for a while. This is very disturbing

Many ssh clients have a keep alives feature that keeps ssh connections alive.

But I don't find that feature in WindTerm, how do I keep ssh connected?

**Comment by kingToolbox** [comment, marked as answer]
Please do the following two operations to set `keepalive` to see if this problem can be solved.

- Check the `Session Setting Dialog - SSH - Connection - Enable TCP keepalives (SO_KEEPALIVE option)`.
- Set `Session Setting Dialog - SSH - Connection - Sends null packet to keep session alive` to nonzero value, like 5 seconds.

If you still have problems, please let me know, thank you!

**Reply by isPoto** [reply]
Thank you for developing such a great SSH client!

**Comment by TaylorChen** [comment]
it's work

**Comment by iTomxy** [comment]
Seems that each session has to be set individually. Is there a way to set this globally ?

---

[#742](https://github.com/kingToolbox/WindTerm/issues/742#issue-1231351988)

**Reply by str1ngthen** [reply]
yes，please click -first choice - Default Session Settings... and same operation
